site/src/sphinx/advanced-structured-logging.rst

```diff
@@ -148,6 +152,70 @@ automatically for you:
         }
     }
 
+Enabling content previews
+-------------------------
+Armeria provides the ``requestContentPreview`` and ``responseContentPreview`` properties in :api:`RequestLog`
+to retrieve the textual representation of the first N bytes of the request and response content.
+However, the properties are disabled by default so they always return ``null`` due to performance overhead.
+You can enable it when you configure :api:`Server`, :api:`VirtualHost` or :api:`Client`.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.server.ServerBuilder;
+    import com.linecorp.armeria.server.VirtualHostBuilder;
+
+    ServerBuilder sb = new ServerBuilder();
+    ...
+    // Enable previewing the content with the maxium length of 100 for textual content.
+    sb.contentPreview(100);
+    ...
+    VirtualHostBuilder vhb = new VirtualHostBuilder("http://example.com");
+    // In this case, the property of virtual host takes precedence over that of server.
+    vhb.contentPreview(150);
+    ...
+    sb.virtualHost(vhb.build());
+
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.ClientBuilder;
+    import com.linecorp.armeria.client.HttpClientBuilder;
+
+    ClientBuilder cb = new HttpClientBuilder();
+    ...
+    cb.contentPreview(100);
+
+Note that the properties are enabled only for textual contents and
+Armeria considers the following contents as textual contents.
+
+- when its type matches ``text/*`` or ``application/x-www-form-urlencoded``.
+- when its charset has been specified. e.g. application/json; charset=utf-8.
+- when its subtype is ``xml`` or ``json``. e.g. application/xml, application/json.
+- when its subtype ends with ``+xml`` or ``+json``. e.g. application/atom+xml, application/hal+json
+
+When a request or response begins, the :api:`ContentPreviewerFactory` set by a user
+creates a :api:`ContentPreviewer` based on HTTP headers to produce the preview when a request or response ends.
+
+You can configure :api:`Server`, :api:`VirtualHost`, or :api:`Client` to use your own :api:`ContentPreviewerFactory`
+and :api:`ContentPreviewer`. e.g.
```

Comment:
I think these two paragraphs need some reorganization. How about explaining first to tell how to enable the binary preview first using the example below and then how to write a completely new previewer (and its factory) from scartch? (so that the flow from your paragraph about textual previews above feels more natural)